packages/augur-sdk/src/state/db/ZeroXOrders.ts

```diff
@@ -0,0 +1,125 @@
+// 2. To cache market orderbooks so a complete pull isnt needed on every subsequent load. We can do this on demand if the full sync above is too slow
+
+export interface OrderData {
+  market: string;
+  price: string;
+  outcome: string;
+  orderType: string;
+  kycToken: string;
+}
+
+export interface Document extends BaseDocument {
+  blockNumber: number;
+}
+
+export interface SnapshotCounterDocument extends BaseDocument {
+  snapshotCounter: number;
+}
+
+export interface StoredOrder extends OrderData {
+  orderHash: string,
+  signedOrder: SignedOrder,
+  amount: string,
+}
+
+/**
+ * Stores 0x orders
+ */
+export class ZeroXOrders extends AbstractDB {
+  protected syncStatus: SyncStatus;
+  protected stateDB: DB;
+  private augur: Augur;
+
+  constructor(
+    db: DB,
+    networkId: number,
+    augur: Augur
+  ) {
+    super(networkId, db.getDatabaseName("ZeroXOrders"), db.pouchDBFactory);
+    this.syncStatus = db.syncStatus;
+    this.stateDB = db;
+    this.augur = augur;
+    // TODO: add indicies
+  }
+
+  static async create(db: DB, networkId: number, augur: Augur): Promise<ZeroXOrders> {
+    const zeroXOrders = new ZeroXOrders(db, networkId, augur);
+    await zeroXOrders.deleteOld();
+    await zeroXOrders.subscribeToMeshEvents();
+    return zeroXOrders;
+  }
+
+  async subscribeToMeshEvents(): Promise<void> {
+    return await this.augur.zeroX.subscribeToMeshEvents(this.handleMeshEvent.bind(this));
+  }
+
+  async deleteOld(): Promise<void> {
+    const oldSnapshotRawDocs = await this.allDocs();
+    const oldSnapshotDocs = oldSnapshotRawDocs.rows ? oldSnapshotRawDocs.rows.map(row => Object.assign(row.doc, { _deleted: true})) : [];
+    if (oldSnapshotDocs.length > 0) {
+      await this.bulkUpsertUnorderedDocuments(oldSnapshotDocs);
+    }
+  }
+
+  // TODO: Investigate actual data returned to see if we need to handle the "KIND" field. If for example a "EXPIRED" kind does not also set the fillableTakerAssetAmount to 0 then we'll need to handle that
+  handleMeshEvent(orderEvents: OrderEvent[]): void {
+    const documents = _.map(orderEvents, this.processOrder.bind(this));
+    this.bulkUpsertUnorderedDocuments(
+      documents
+    ).then((success) => {
+      if (success) {
+        augurEmitter.emit("ZeroXOrders", documents);
+      } else {
+        throw new Error('Unable to handle mesh events for ZeroX Orders');
+      }
+    });
+  }
+
+  async sync(): Promise<void> {
+    const orders: Array<OrderInfo> = await this.augur.zeroX.getOrders();
+    let success = true;
+    let documents;
+    if (orders.length > 0) {
+      documents = _.map(orders, this.processOrder.bind(this));
+
+      success = await this.bulkUpsertUnorderedDocuments(
+        documents
+      );
+    }
+    if (success) {
+      augurEmitter.emit("ZeroXOrders", {});
```

Comment:
Why not include the docs in the payload here like we are doing above?

Reply:
The context here is a full sync. This will always be every single order on Augur.